### DISCUSSION TITLE
Artworks on MPCFills have disappeared

### DISCUSSION THREAD

**Discussion by TrueBlue726** [chilli-axe/mpc-autofill#220] [General]
![MPCFILL ERROR-1](https://github.com/chilli-axe/mpc-autofill/assets/95557492/6a0b761b-9500-40de-85c5-4d0f004898e1)
![MPCFILL ERROR-2](https://github.com/chilli-axe/mpc-autofill/assets/95557492/d695b28f-be05-4610-8358-e471d892fd8b)
I was working on a MPC order on MPCFIll when I noticed that the artworks for Orcish Army token, which I had access to just 2 days prior, have disappeared.  In a few minutes, even more artworks have disappeared and now I cannot complete my order.  Please fix this issue ASAP.  I want to play a large order today but now I can't.  Thank you.

**Comment by ndepaola** [comment]
g'day. for context, we don't store image thumbnails in the backend server directly, but rather we stream thumbnails directly from google drive on-demand. image links are formatted like https://drive.google.com/thumbnail?sz=w800-h800&id=1LrVX0pUcye9n_0RtaDNVl2xPrQgn7CYf - we use the same endpoint for the small thumbnails shown through most of the app and the large thumbnails shown in the detailed view modal.

we've known for a while that google applies rate limiting to this endpoint, but up until now the upper limit has been generous enough to rarely bother users. the problem should go away for you if you pause and come back in a few hours or the next day.

it seems that google has recently (in the last few days) made this upper limit more strict, so more users have been reporting this issue. i haven't seen any official notice from google about a change like this. nothing has mechanically changed about how we serve images in years.

to me it seems that the correct solution is to store thumbnails in a CDN, but there are several complexities to this task and i haven't commenced work on the technical design yet, so there's no ETA on when this will be fixed. i've also considered serving images from the google drive API but it does not appear to have functionality for serving images at different resolutions (which is a strict requirement for us).

this project is fuelled by my unpaid labour and that factors heavily into how quickly meaty issues like this can be turned around. for now, i suggest you try again later today or tomorrow, when google's rate limit on your IP address has worn off.

**Comment by TrueBlue726** [comment]
Hello Nicholas,

Understood and thank you very much for your hard work.

**Comment by MangOTigeR22** [comment]
Seems this issue has continued to increase in frequency. I have noticed it not displaying images for a couple of days now after trying at different points through the day. Might get one image here or there.

**Comment by TrueBlue726** [comment]
Same for me too.  I want to place another order through MPC Fill but it
hasn't been easy.

**Comment by ndepaola** [comment]
this is fixed now (i just haven't PR'd my cloudflare worker into the repo yet)
